packages/core/streams/parsers.ts

```diff
@@ -0,0 +1,59 @@
+import { AIStreamMode, AIStreamParser } from './types'
+import { trimStartOfStreamHelper } from './utils'
+
+export type AIStreamParsers = (mode: AIStreamMode) => AIStreamParser | undefined
+
+export const openaiParsers: AIStreamParsers = mode => {
+  const trimStartOfStream = trimStartOfStreamHelper()
+
+  switch (mode) {
+    case 'raw':
+      return
+
+    case 'text':
+      return data => {
+        // TODO: Needs a type
+        const json = JSON.parse(data)
+
+        // this can be used for either chat or completion models
+        const text = trimStartOfStream(
+          json.choices[0]?.delta?.content ?? json.choices[0]?.text ?? ''
+        )
+
+        return text
+      }
+  }
+}
```

Comment:
We can easily add JSON parsing on completions for integration with [OpenAI function calling](https://openai.com/blog/function-calling-and-other-api-updates) (or manual equivalents, e.g. `Respond only with valid JSON`-like prompts) by updating `type AIStreamMode = "raw" | "text" | "json"` and adding the OpenAI and Anthropic parsers.